File: apps/mybookkeeper/backend/app/schemas/tax/tax_return.py
import uuid
from datetime import datetime

from pydantic import BaseModel

TAX_DISCLAIMER = (
    "Tax values are AI-generated estimates. Review all values before filing. "
    "You are responsible for the accuracy of your tax return."
)


class TaxReturnRead(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    tax_year: int
    filing_status: str
    jurisdiction: str
    status: str
    needs_recompute: bool
    filed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    disclaimer: str = TAX_DISCLAIMER

    model_config = {"from_attributes": True}


class TaxReturnCreate(BaseModel):
    tax_year: int
    filing_status: str = "single"
    jurisdiction: str = "federal"

File: apps/mybookkeeper/frontend/src/shared/types/tax/tax-return.ts
export type TaxReturnStatus = "draft" | "ready" | "filed";

export interface TaxReturn {
  id: string;
  organization_id: string;
  tax_year: number;
  filing_status: string;
  jurisdiction: string;
  status: TaxReturnStatus;
  needs_recompute: boolean;
  filed_at: string | null;
  created_at: string;
  updated_at: string;
}
